lib/identify.go

```diff
@@ -29,12 +50,54 @@ func IdentifyRequest(ctx context.Context, maStr string) (*IdentifyInfo, error) {
 	}
 
 	if err := h.Connect(ctx, *ai); err != nil {
-		return nil, err
+		if !usingBogusPeerID {
+			return nil, err
+		}
+		newPeerId, err := extractPeerIDFromError(err)
+		if err != nil {
+			return nil, err
+		}
+		ai.ID = newPeerId
+		if err := h.Connect(ctx, *ai); err != nil {
+			return nil, err
+		}
 	}
 
 	return extractIdentifyInfo(h.Peerstore(), ai.ID)
 }
 
+func extractPeerIDFromError(inputErr error) (peer.ID, error) {
```

Comment:
Note: this function is no fun and is brittle. Suggestions welcome for improvements.